sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -1182,18 +1183,24 @@ private static boolean hasTimestampAnnotation(List<Annotation> annotations) {
     errors.checkArgument(void.class.equals(m.getReturnType()), "Must return void");
 
     Type[] params = m.getGenericParameterTypes();
-    errors.checkArgument(params.length == 3, "Must have exactly 3 arguments");
+    errors.checkArgument(params.length == 4, "Must have 4 arguments");
```

Comment:
If we expect a specific set of parameters and order, can we upgrade this error message to explicitly state it? I'm imagining writing this method signature and hitting each of these errors one at a time to deduce the correct signature.